rb-files/decorator.rb

```diff
@@ -0,0 +1,23 @@
+require './nameable'
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
```

Comment:
- [ ] For best practices and to match the project requirements, I think it would be better if you define `setter` and `getter` on the `nameable` attribute as they provide encapsulation, control over attribute access, abstraction, code organization, and improved readability, making them important tools for writing maintainable and robust object-oriented code in Ruby.
```ruby
...
attr_accessor :nameable
....
```